### ISSUE TITLE
Transfer of project ownership results in duplicated user, owner privileges inaccessible

### ISSUE THREAD

**Issue by bob-reus** [FlutterFlow/flutterflow-issues#3186]
### Can we access your project?

- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

Attempting to transfer the the ownership from one user to the other results in a duplicate entry of the recipient of ownership status; one with Editor privileges, and one with Owner privileges. 

![ff-ownerbug](https://github.com/FlutterFlow/flutterflow-issues/assets/111222628/5f705010-973a-4662-aa8f-844057144f03)

In the above screenshot is the result of attempting a transfer from Vincent to Bob, as described by the docs, seen from Bob's account. 

This results in a deadlock-like situation where the 'old' owner lost their privileges, but the 'new' owner (me) doesn't get them, because they're apparently assigned to a duplicate account? It also means that there is no way to remove this duplicated account. 

### Expected Behavior

A transfer of ownership [as described in the documentation.](https://docs.flutterflow.io/settings-and-integrations/project-setup/collaboration#transfer-ownership-of-a-project-to-another-user)

### Steps to Reproduce

1. Navigate to Settings & Integrations (from the left Navigation Menu) > Project Setup > Collaboration > Invite Users.
2. Click on the Status (e.g., Active) of the person you want to transfer project ownership to. You can only transfer ownership to someone with a status of Active.
3. A popup will appear, and select Yes. 

[As described in the docs.](https://docs.flutterflow.io/settings-and-integrations/project-setup/collaboration#transfer-ownership-of-a-project-to-another-user)

### Reproducible from Blank

- [ ] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

IT40jvHZvI9inMMF0bn+VO5H+SkXJW4mU783l9VBaCosfZT2BLMcec/RVERBZ7zmTGx6e06khWoW3NbyvtnTJfUnHy2VfaJJ+Lp+VgLKUT+XMM2pFJWgZ3wlP9tWCXKH0J23hyZSIbFsR1Zl7FyEAPCWcB/aCZTxDEYwDq/LZO4=

### Visual documentation

![ff-ownerbug](https://github.com/FlutterFlow/flutterflow-issues/assets/111222628/5f705010-973a-4662-aa8f-844057144f03)

I'm not going to attempt reproducing this. 

### Environment

```markdown
- FlutterFlow version: v4.1.61+
- Platform: platform independent
- Browser name and version: Chromium Version 125.0.6422.60 (Official Build) Arch Linux (64-bit)
- Operating system and version affected: Windows 11, Linux (BTW I Use Arch)
```

### Additional Information

We lost ownership over our project.

**Comment by rzambroni** [comment]
Hey @bob-reus, thanks for the report, and sorry about the issue, it is quite strange to be honest. I tested this my self on other projects and i was not able to replicate. I was able to transfer the project to a new owner and transfer it back to my main account afterwards.

For this particular case i would have to ask you to reach out to our support team, so they can help you fix the issue on your account. 

Do you mind following up with them? You can do that either by email: [support@flutterflow.io](mailto:support@flutterflow.io) or using the in-app chat inside of FlutterFlow.

Thanks!

**Comment by bob-reus** [comment]
I've contacted support, and we haven't reached a satisfying conclusion (yet, but we'll get there :) ), but just now I did discover some more specifics:

-> When I delete all site data and cache, and log in again, ownership transfer is successful on the main branch.
-> If I then want to switch to a different branch, I have to log in again
-> After I log in again, the situation is as described in the initial bug report

So, without having any knowledge of what happens behind the curtain, it seems that there's a separate (or additional) set of user settings for branches, which don't transfer over as well. 

I am not sure if this is intended behavior, but it's certainly unintuitive.
